src/StableHashTraits.jl

```diff
@@ -324,41 +326,128 @@ See also: [`StableHashTraits.hash_method`](@ref).
 write(io, x, context) = write(io, x)
 write(io, x) = Base.write(io, x)
 
-function stable_hash_helper(x, hash_state, context, ::WriteHash)
+function stable_hash_helper(x, hash_state, context, root, ::WriteHash)
     return update_hash!(hash_state, x, context)
 end
 
 #####
-##### IterateHash 
+##### FnHash 
 #####
 
-struct IterateHash end
-function stable_hash_helper(xs, hash_state, context, ::IterateHash)
-    return hash_foreach(hash_state, context, xs) do x
-        return x, hash_method(x, context)
+struct FnHash{F,H}
```

Comment:
This block of changes is simply moved around code. It is the same block that existed in the previous version.